docs/ref/checks.txt

```diff
@@ -141,6 +141,13 @@ Models
 * **models.E022**: ``<function>`` contains a lazy reference to
   ``<app label>>.<model>``, but app ``<app label>`` isn't installed or
   doesn't provide model ``<model>``.
+* **models.E023**: The model name ``<model>`` starts or ends with underscores.
```

Comment:
"... cannot start or end with an underscore as it collides with the relational lookup syntax." (or something like that)
The message here should be the same as in the check.